test/stdlib/KVOKeyPaths.swift

```diff
@@ -108,7 +108,72 @@ print("target removed")
 // CHECK-NEXT: swiftValue 42, objcValue three
 // CHECK-NEXT: swiftValue 42, objcValue four
 // CHECK-NEXT: swiftValue 13, objcValue four
+// The next 2 logs are actually a bug and shouldn't happen
 // CHECK-NEXT: swiftValue 13, objcValue four
 // CHECK-NEXT: swiftValue 13, objcValue four
 // CHECK-NEXT: target removed
 
+//===----------------------------------------------------------------------===//
+// Test NSKeyValueObservingCustomization issue with observing from the callbacks
+//===----------------------------------------------------------------------===//
+
+class Target2 : NSObject, NSKeyValueObservingCustomization {
+    @objc dynamic var name: String?
+
+    class Dummy : NSObject {
+        @objc dynamic var name: String?
+    }
+
+    // In both of the callbacks, observe another property with the same key path.
+    // We do it in both because we're not sure which callback is invoked first.
+    // This ensures that using KVO with key paths from one callback doesn't interfere
+    // with the ability to look up the key path using the other.
+    static func keyPathsAffectingValue(for key: AnyKeyPath) -> Set<AnyKeyPath> {
+        print("keyPathsAffectingValue: key == \\.name:", key == \Target2.name)
+        withExtendedLifetime(Dummy()) { (dummy) in
+			_ = dummy.observe(\.name) { (_, _) in }
+		}
+        return []
+    }
+
+    static func automaticallyNotifiesObservers(for key: AnyKeyPath) -> Bool {
+        print("automaticallyNotifiesObservers: key == \\.name:", key == \Target2.name)
+        withExtendedLifetime(Dummy()) { (dummy) in
+			_ = dummy.observe(\.name) { (_, _) in }
+		}
```

Comment:
These 2 lines have hard tabs instead of spaces.

Reply:
… one day, humanity will solve the problem of varying configurations between hosts and contexts